### PR TITLE
Fixes #263 - Added fetching and rendering of data from the backend to the frontend

### LINKED ISSUE #263: Update frontend to fetch() post data from the web server
In #262 we're adding routes to the web server to allow for getting `data from the database, /posts` (all post IDs in the database); and `/post/<id>` (get a particular post).  In #261 we're updating the frontend code to make use of all the data in a post (see #259).

We also need to change the frontend code so that it uses [fetch()](https://developers.google.com/web/updates/2015/03/introduction-to-fetch) to get post JSON data from the server, and "renders" it in our frontend.

For now, we can do this with jQuery or raw DOM API calls, and later we can move to some framework and components.

Keep the scope of this fix small: we need to get data from the web server, and use it to show posts in the page.

### PR DESCRIPTION
**Pull request by lucacataldo**
Added the fetching & rendering of posts in frontend. This includes fetching post & participant data using jQuery (currently dummy data in src/frontend/dummy_data),  and rendering it using a templated <article> element. I picked some posts from the current planet that included elements I though would bring up some edge cases and adjusted CSS accordingly so that when this gets hooked into the actual routes it'll be a bit less garbled and legible. Also, since pagination.js dialogue.js are no longer relevant or being used, I removed them.

### CODE REVIEW

**Comment by cindyledev** [comment]
I also think everything in `main.css` can be a separate PR by itself as well

File: src/frontend/dummy_data/participants.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
+{
+  "participants": [
+    {
+      "name": "John Smith",
+      "url": "https:google.ca"
```
Comment (humphd) [inline src/frontend/dummy_data/participants.json]: Let's use proper URLs throughout.

File: src/frontend/dummy_data/posts.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "NOTE": "THIS IS TEMPORARY DUMMY DATA UNTIL DATA IS EXPOSED BY SERVER",
```
Comment (humphd) [inline src/frontend/dummy_data/posts.json]: Remove this line

File: src/frontend/dummy_data/posts.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      "updated": "October 15, 2019",
```
Comment (humphd) [inline src/frontend/dummy_data/posts.json]: Can you make these proper JSON style date strings?

File: src/frontend/dummy_data/posts.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      "published": "October 15, 2019",
+      "url": "https://neilong31.wordpress.com/2019/09/13/web-based-note-taker-forevernote/",
+      "site": "https://neilong31.wordpress.com/",
+      "guid": "null"
```
Comment (humphd) [inline src/frontend/dummy_data/posts.json]: Should be something other than null.  Our guids look like `"tag:blogger.com,1999:blog-3586542501367863430.post-1706207209862339192"`, basically a URI of some type.

File: src/frontend/index.html [hunk 5]
```diff
@@ -54,216 +60,111 @@
...
-          nav.style.boxShadow = '0px 0px 0px var(--box-shadow-color)';
-        };
-      };
+      // Toggle the mobile menu
+      $('#mobile-burger').click(() => {
+        const nav = $('nav');
+
+        nav.css({
+          transform: 'translateX(-200px)',
+          boxShadow: '0px 0px 100px var(--box-shadow-color)',
```
Comment (humphd) [inline src/frontend/index.html]: CSS in `.css` files please.  Let's use classes if possible.

File: src/frontend/index.html [hunk 6]
```diff
@@ -54,216 +60,111 @@
...
+        });
+
+        const close = $('#mobile-burger-close');
+
+        close.click(function() {
+          nav.css({
+            transform: 'translateX(0px)',
+            boxShadow: '0px 0px 0px var(--box-shadow-color)',
```
Comment (humphd) [inline src/frontend/index.html]: Same thing here.

File: src/frontend/index.html [hunk 7]
```diff
@@ -54,216 +60,111 @@
...
-      // CODE TO TOGGLE THE MOBILE MENU DRAWER
-      document.querySelector('#mobile-burger').onclick = function(e) {
-        let nav = document.querySelector('nav');
-        nav.style.transform = 'translateX(-200px)';
-        nav.style.boxShadow = '0px 0px 100px var(--box-shadow-color)';
+    <!-- SCRIPTS -->
 
-        let close = document.querySelector('#mobile-burger-close');
+    <script>
+      $(() => {
```
Comment (humphd) [inline src/frontend/index.html]: JavaScript in `.js` files please.

File: src/frontend/style/main.css [hunk 8]
```diff
@@ -71,7 +71,7 @@ nav a:visited {
   display: inline-block;
   width: 150px;
   background: var(--nav-btn-bgcolor);
-  font-size: 1.25em;
+  font-size: 1em;
```
Comment (humphd) [inline src/frontend/style/main.css]: Can these CSS changes get done in another PR?  They seem unrelated to this fix.

**Comment by lucacataldo** [comment]
Thanks for the feedback @cindyledev @miggs125 @humphd its greatly appreciated! Will be putting up a new commit in the coming days addressing the issues brought up. 

In regards to having the Javascript in separate files, I had all the code separated into modular files initially (UI tools in one, fetching in another, etc.) but was getting a lot of eslint errors where it was picking up variables as unused between files because they weren't really linked anywhere other than the <script> tags in the HTML. Wasn't able to fix that so would it be alright to have everything in one (separate) .js file for now and maybe have someone file an issue / PR to make things more modular and correctly integrated or adjust eslint settings in the future?

**Comment by lucacataldo** [comment]
So, I've got a commit waiting and ready to go with all the fixes specified including dropping the CSS changes. I previously added these to address UX / UI issues that arose when I added dummy data to better represent typical posts. The following is a screenshot of a mobile view with the proposed changes:

image

As you can see this will greatly impact the way that posts are rendered. 


@humphd @miggs125 @cindyledev 
I'll leave it up to you guys as to whether or not to include them or not since I can personally see valid arguments for both including and not including them. 🤷‍♂ 

I think everything else that was mentioned has been fixed and will put up the PR when I hear back regarding the CSS changes. Thanks again for taking the time to look things over!

**Comment by humphd** [comment]
@lucacataldo the discussions of splitting this up aren't aimed at having you throw changes away; rather, we're trying to improve the reviewability of the work, and isolate changes to their own PRs.  Doing the fixes to CSS is important, and we need that.  But the question is: do we need it in the same PR as one which adds data fetching?

If it's not possible for you to separate these, then please go ahead and do it all at once.  But in future, I want all of us to try and think about how to keep our changes simple, focused on doing only 1 thing, and making it easier for them to get reviewed, tested, and merged.

**Comment by lucacataldo** [comment]
@humphd totally understandable and it was a mistake on my part to go down the rabbit hole so to speak. Will definitely keep this in mind in the future. As for now, I am going to give it a shot to minimize the amount of CSS added in this PR and file a new issue for the other smaller fixes.

File: src/frontend/dummy_data/participants.json (hunk 1 above)
Comment (lucacataldo) [reply src/frontend/dummy_data/participants.json]: Good catch, thanks!

File: src/frontend/dummy_data/posts.json (hunk 3 above)
Comment (lucacataldo) [reply src/frontend/dummy_data/posts.json]: Good idea, I've now used the Date object methods to turn a properly formatted string while using the correct JS formatted Timestamp to create the Date object.

File: src/frontend/index.html (hunk 5 above)
Comment (lucacataldo) [reply src/frontend/index.html]: Added classes, now using jQuery .addClass and .removeClass methods to swap states.

File: src/frontend/index.html (hunk 7 above)
Comment (lucacataldo) [reply src/frontend/index.html]: Moved to main.js, would appreciate any resources on how to keep code in the front end in modular files as I was struggling with that

**Comment by lucacataldo** [comment]
@cindyledev @humphd @miggs125 changes have been pushed, have a look and let me know if there's anything that's still outstanding!

File: package.json [hunk 9]
```diff
@@ -23,8 +23,11 @@
     "pretest": "npm run lint",
     "test": "npm run jest",
     "jest": "cross-env MOCK_REDIS=1 jest",
-    "jest-watch":"cross-env MOCK_REDIS=1 jest --watch",
-    "start": "node src",
+    "jest-watch": "cross-env MOCK_REDIS=1 jest --watch",
+    "start": "node src/backend",
```
Comment (humphd) [inline package.json]: None of these files should be changed in your branch as far as I can tell.  We've had people overwrite these already in the past few days, I'm not eager to repeat it.  If I'm wrong and you do need these to be updated, let me know why.

**Comment by humphd** [comment]
@lucacataldo see my work in #426, which takes what you have here, and connects it into the new backend we just merged.  You could use that to update your code in this PR.

**Comment by humphd** [comment]
Closing and continuing in my other PR with new changes added.
